### DISCUSSION TITLE
关于 ProxyPin 默认根证书的问题

### DISCUSSION THREAD

**Discussion by Fluair** [wanghongenpin/proxypin#267] [Q&A] [answered]
“ProxyPin默认根证书对所有用户都是相同的” 这句话是否意味着 ProxyPin 默认根证书的私钥是公开的？我想是的。
如此的话将 ProxyPin 默认根证书安装到系统里是十分危险的。
还好最新版本可以选择生成随机根证书，因此在安装根证书之前应该先随机生成根证书，替代不安全的默认证书。

我想 ProxyPin 能否在安装后自动生成根证书而不是使用自带的默认证书？

**Comment by wanghongenpin** [comment, marked as answer]
对于小白会增加困难度，因为手机和电脑证书不一样，都需要安装下，以及证书模块都需要修改。
而且只有开启抓包之后才会用自定义证书，手机同时只能开启一个VPN，其他应用拦截不了，电脑同样只能设置一个代理。
对于熟悉抓包证书机制的，多了下重新生成证书操作，不是什么大问题。
